### DISCUSSION TITLE
Attributing a destructured object to an object to spread it, is it possible?

### DISCUSSION THREAD

**Discussion by j3rrb** [nodejs/help#4865] [Q&A] [answered]
I'm trying to destructure an object and assign it to another object to spread it, is it possible or already exists another way to do that?
If not, it would be good if it can be done like: const obj = {a: 1, b: 2, c: 3}; const a = {a, b} = obj;

**Comment by aduh95** [comment, marked as answer]
Here's how you can do that:

```js
const obj = {a: 1, b: 2, c: 3};
const { a, b } = obj;
const newObj = { a, b };
```

or

```js
const obj = {a: 1, b: 2, c: 3};
const newObj = { a: obj.a, b: obj.b };
```

Note that Node.js doesn't have control over JavaScript the language, discussion over the language spec is happening in https://github.com/tc39/proposals.

**Reply by aaarichter** [reply]
if you want to omit known keys, another way would be
```js
const { c, ...newObj } = obj;
```
`newObj` will be `{a: 1, b: 2}`
